drivers/adodb-postgres64.inc.php

```diff
@@ -958,6 +961,7 @@ class ADORecordSet_postgres64 extends ADORecordSet{
 	var $_blobArr;
 	var $databaseType = "postgres64";
 	var $canSeek = true;
+	var $adodbFetchMode;
```

Comment:
Ah! I saw that but clearly misread what it stated, and overlooked the duplicate due to the different variable name. I'll fix that this evening.

Reply:
No worries. Cheers 😃

Reply:
Upon closer inspection, I think this would actually be a bit more involved, and would be a breaking change requiring much more work than it appears at a glance

Currently, the contents of adodbFetchMode are the generic ADOdb fetch mode type (for example, ADODB_FETCH_ASSOC), while fetchMode has the PostgreSQL-specific fetch mode (ie: PGSQL_ASSOC).

I _could_ crunch it down to one or the other, but if I do that, which one would it be best to keep? In trying to figure that out, I found the following in an old changelog explaining why this property exists:

> FetchMode cached in recordset is sometimes mapped to native db fetchMode. Normally this does not matter, but when using cached recordsets, we need to switch back to using adodb fetchmode. So we cache this in $rs->adodbFetchMode if it differs from the db's fetchMode.

The other instances of adodbFetchMode in the code:

1. An instance here, which appears to exist for working with records returned by the other drivers that use this property: https://github.com/ADOdb/ADOdb/blob/eb6d8442e157afd761e4dbcbbda0243235a6105b/adodb-csvlib.inc.php#L77

2. Instances with the exact same purpose as in this PostgreSQL driver in the following drivers: PDO, SQLite, SQLite3, OCI8, MySQLi

Either way, I think the contents of that changelog entry (or something to that effect) should be duplicated as a comment in the locations this property is used to avoid future confusion.

Reply:
Oh, and one more point since I'm thinking about it: Why does ADODB_sqlite still exist at all? As far as I'm aware, the original ext/sqlite extension that driver uses is not available in any version of PHP that ADOdb supports.

Reply:
Just like SQLite, there are many drivers in the system that don't currently work, think informix, netezza, frontbase. While we have documented them as no longer working or supported, the most appropriate time to remove them would be in a major release (think ADOdb 6.0 or something like that). As we haven't got anywhere near doing something like that, they will probably sit out in the drivers directory like old, forgotten friends. They're lightweight and causing no harm so they'll probably sit out there forever...